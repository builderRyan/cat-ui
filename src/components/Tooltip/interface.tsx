export interface StyledTooltipContentProps {
  top?: number;
  left?: number;
}
