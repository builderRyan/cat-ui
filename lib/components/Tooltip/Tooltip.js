"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var classnames_1 = require("classnames");
var __1 = require("../");
require("./index.scss");
var Tooltip = (function (_super) {
    __extends(Tooltip, _super);
    function Tooltip(props) {
        var _this = _super.call(this, props) || this;
        _this.handleChange = function (_a) {
            var _b = _a.popup, popup = _b === void 0 ? false : _b;
            _this.setState({
                popup: popup
            });
        };
        _this.state = {
            popup: false
        };
        return _this;
    }
    Tooltip.prototype.render = function () {
        var _a = this.props, prefixCls = _a.prefixCls, className = _a.className, style = _a.style, placement = _a.placement, title = _a.title, trigger = _a.trigger, children = _a.children;
        var popup = this.state.popup;
        var classes = classnames_1.default(prefixCls + "-pop", className, {
            popuped: popup
        });
        return (React.createElement(__1.Popup, { prefixClx: prefixCls, className: classes, innerCls: 'hi-children', style: style, placement: placement, title: title, trigger: trigger, onChange: this.handleChange }, children));
    };
    Tooltip.defaultProps = {
        prefixCls: 'hmly',
        placement: 'top',
        trigger: 'hover'
    };
    return Tooltip;
}(React.Component));
exports.default = Tooltip;
//# sourceMappingURL=Tooltip.js.map