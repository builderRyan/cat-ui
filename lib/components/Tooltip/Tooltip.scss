.hmly-children {
  margin: 200px auto;
  width: 300px;
  background-color: #f1f3f4;
  text-align: center;
  font-size: 0.875rem;
  min-width: 64px;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 500;
  line-height: 1.5;
  border-radius: 4px;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  padding: 5px;
  cursor: pointer; }
  .hmly-children:hover {
    text-decoration: none;
    background-color: rgba(0, 0, 0, 0.08); }

.hmly-pop {
  opacity: 0;
  color: #fff;
  font-weight: 500;
  line-height: 1.5;
  font-size: 14px;
  border-radius: 3px;
  padding: 3px 10px;
  background-color: #fff;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, opacity 450ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; }
  .hmly-pop.popuped {
    opacity: 1;
    background-color: #6f6f6f; }

