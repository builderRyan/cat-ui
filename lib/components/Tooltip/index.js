"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Tooltip_1 = require("./Tooltip");
exports.default = Tooltip_1.default;
