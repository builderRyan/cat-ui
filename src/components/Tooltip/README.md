# Tooltip

## Example

```jsx
 <Tooltip
    position='top'
    width={200}
    content={<div>tooltip content</div>}
  >
  tooltip
</Tooltip>
```

## API

|   属性    |   类型   |   默认值    |   描述   |
|:---------:|:--------:|:-----------:|:--------:|
| className | `string` |      -      |   类名   |
|   style   | `object` |      -      | 行内样式 |
|   mode    | `string` | 'topCenter' | 方向，可选值 `bottomLeft` `buttonCenter` `bottomRight` `topLeft` `topCenter` `topRight` |
|   width    | `number` | - | 宽度 |
|   content    | `React.ReactNode` | - | 内容 |