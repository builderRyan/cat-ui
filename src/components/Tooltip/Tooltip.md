Example

```jsx
<div className='top-box'>
  <Tooltip placement='topLeft' title={<div>Add</div>}>
    <div className='item top-left'>top-left</div>
  </Tooltip>
  <Tooltip placement='top' title={<div>Add</div>}>
    <div className='item top-left'>top</div>
  </Tooltip>
  <Tooltip placement='topRight' title={<div>Add</div>}>
    <div className='item top-right'>top-right</div>
  </Tooltip>
</div>

<style>
  {`
    .top-box,
    .bottom-box {
      width: 900px;
      margin: 60px auto;
      text-align: center;
    }
    .item {
      display: inline-block;
      font-size: 0.875rem;
      transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
      box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
      border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
      font-family: "Roboto", "Helvetica", "Arial", sans-serif;
      font-weight: 500;
      line-height: 1.5;
      border-radius: 4px;
      letter-spacing: 0.02857em;
      text-transform: uppercase;
      padding: 10px;
      text-align: center;
      cursor: pointer;
      margin: 0 5%;
    }
    .item:hover {
      text-decoration: none;
      background-color: rgba(0, 0, 0, 0.08);
    }
  `}
</style>
```
