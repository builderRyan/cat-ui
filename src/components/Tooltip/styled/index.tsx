import styled from 'styled-components'
import { primaryBtnColor } from '../../../styles/theme'

const StyledTooltip = styled.div``;

const StyledTooltipContent = styled.div`
  position: relative;
  padding: 5px;
  background-color: #eee;
  &.hmly-tooltip-topCenter::after {
    position: absolute;
    top: 30px;
    left: calc(50% - 6px);
    content: " ";
    width: 0;
    height: 0;
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-top: 10px solid #eee;
  }
`;

export { StyledTooltip, StyledTooltipContent }
