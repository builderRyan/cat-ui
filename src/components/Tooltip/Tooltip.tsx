import * as React from 'react'
import { Component } from 'react'
import classNames from 'classnames'
import { StyledTooltip, StyledTooltipContent } from './styled'
import Popover from '../Popover'

export interface TooltipProps {
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: object;
  /** Tooltip hover后显示的内容 */
  mode?: 'hover' | 'click';
  /** 定位的方向 */
  position?: 'bottomLeft' | 'topCenter';
  /** Tooltip hover后显示的内容 */
  content?: string | React.ReactNode;
}

/**
 * 文字提示
 */
class Tooltip extends Component<TooltipProps, any> {
  private contentRef: any;
  static defaultProps = {
    mode: 'hover',
    position: 'topCenter'
  }

  constructor (props: TooltipProps) {
    super(props)
    this.state = {
      isPopOpen: false,
      tipTop: 0,
      tipLeft: 0,
    }
    this.contentRef = React.createRef()
  }

  componentDidUpdate () {
    const contentDOM = this.contentRef.current
    if (!contentDOM) { return }
    const { isPopOpen, tipTop, tipLeft } = this.state
    const contentRect = contentDOM.getBoundingClientRect()
    if (isPopOpen && tipTop === 0 && tipLeft === 0) {
      this.setState({
        tipTop: contentRect.height,
        tipLeft: contentRect.width / 2 - 8
      })
    }
  }

  onPopoverChange = (value: boolean) => {
    this.setState({ isPopOpen: value })
  }

  render() {
    const { className, style, mode, content, position, children } = this.props
    const { isPopOpen, tipTop, tipLeft } = this.state
    const triggerClass = classNames('hmly-tooltip-trigger', className)
    const contentClass = classNames('hmly-tooltip-content', {
      [`hmly-tooltip-${position}`]: position
    })

    return (
      <Popover
        visible={isPopOpen}
        mode={mode}
        position={position}
        onChange={this.onPopoverChange}>
        <Popover.Trigger>
          <StyledTooltip
            className={triggerClass}
            style={style}>
            {children}
          </StyledTooltip>
        </Popover.Trigger>
        <Popover.Content>
          <StyledTooltipContent
            ref={this.contentRef}
            className={contentClass}
            top={tipTop}
            left={tipLeft}>
            {content}
          </StyledTooltipContent>
        </Popover.Content>
      </Popover>

    )
  }
}

export default Tooltip
