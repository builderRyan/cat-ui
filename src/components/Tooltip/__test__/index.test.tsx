import React from 'react';
import Enzyme from 'enzyme';
import { shallow, mount } from 'enzyme';
// import * as Adapter from 'enzyme-adapter-react-16';
import Tooltip from '../index';

// Enzyme.configure({ adapter: new Adapter() });

describe('Tooltip', () => {
  test('renders correctly', () => {
    const wrapper1 = shallow(
      <Tooltip content='tooltip content' position='top'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper2 = shallow(
      <Tooltip content='tooltip content' position='topLeft'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper3 = shallow(
      <Tooltip content='tooltip content' position='topRight'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper4 = shallow(
      <Tooltip content='tooltip content' position='right'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper5 = shallow(
      <Tooltip content='tooltip content' position='rightTop'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper6 = shallow(
      <Tooltip content='tooltip content' position='rightBottom'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper7 = shallow(
      <Tooltip content='tooltip content' position='bottom'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper8 = shallow(
      <Tooltip content='tooltip content' position='bottomLeft'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper9 = shallow(
      <Tooltip content='tooltip content' position='bottomRight'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper10 = shallow(
      <Tooltip content='tooltip content' position='left'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper11 = shallow(
      <Tooltip content='tooltip content' position='leftTop'>
        <span>trigger</span>
      </Tooltip>
    );
    const wrapper12 = shallow(
      <Tooltip content='tooltip content' position='leftBottom'>
        <span>trigger</span>
      </Tooltip>
    );
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper3).toMatchSnapshot();
    expect(wrapper4).toMatchSnapshot();
    expect(wrapper5).toMatchSnapshot();
    expect(wrapper6).toMatchSnapshot();
    expect(wrapper7).toMatchSnapshot();
    expect(wrapper8).toMatchSnapshot();
    expect(wrapper9).toMatchSnapshot();
    expect(wrapper10).toMatchSnapshot();
    expect(wrapper11).toMatchSnapshot();
    expect(wrapper12).toMatchSnapshot();
  });

  test('When you click trigger, the content should be displayed.', () => {
    const wrapper = mount(
      <Tooltip mode='click' content='tooltip content' position='top'>
        <span className='my-trigger'>trigger</span>
      </Tooltip>
    );
    wrapper.find('.my-trigger').simulate('click');
    let popoverContentDOM = document.querySelector(
      '.cat-tooltip__content--top'
    );
    expect(popoverContentDOM instanceof HTMLElement).toEqual(true);
  });

  test('When you click trigger, the content should be displayed', () => {
    const wrapper = mount(
      <Tooltip mode='click' content='tooltip content' position='top'>
        <span className='my-trigger'>trigger</span>
      </Tooltip>
    );
    wrapper.find('.my-trigger').simulate('click');
    const popoverContentDOM = document.querySelector(
      '.cat-tooltip__content--top'
    );
    expect(popoverContentDOM instanceof HTMLElement).toEqual(true);
  });
  
  test('When you mouseover trigger, the content should be displayed', () => {
    const wrapper = mount(
      <Tooltip mode='click' content='tooltip content' position='top'>
        <span className='my-trigger'>trigger</span>
      </Tooltip>
    );
    wrapper.find('.my-trigger').simulate('mouseover');
    const popoverContentDOM = document.querySelector(
      '.cat-tooltip__content--top'
    );
    expect(popoverContentDOM instanceof HTMLElement).toEqual(true);
  });
});
