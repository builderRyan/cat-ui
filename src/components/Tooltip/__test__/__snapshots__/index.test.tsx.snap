// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly 1`] = `
<Popover
  contentPosition={
    Array [
      "center",
      "bottom",
    ]
  }
  mode="hover"
  offsetX={0}
  offsetY={-10}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "center",
      "top",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 2`] = `
<Popover
  contentPosition={
    Array [
      "left",
      "bottom",
    ]
  }
  mode="hover"
  offsetX={0}
  offsetY={-10}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "left",
      "top",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 3`] = `
<Popover
  contentPosition={
    Array [
      "right",
      "bottom",
    ]
  }
  mode="hover"
  offsetX={0}
  offsetY={-10}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "right",
      "top",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 4`] = `
<Popover
  contentPosition={
    Array [
      "left",
      "center",
    ]
  }
  mode="hover"
  offsetX={10}
  offsetY={0}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "right",
      "center",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 5`] = `
<Popover
  contentPosition={
    Array [
      "left",
      "top",
    ]
  }
  mode="hover"
  offsetX={10}
  offsetY={0}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "right",
      "top",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 6`] = `
<Popover
  contentPosition={
    Array [
      "left",
      "bottom",
    ]
  }
  mode="hover"
  offsetX={10}
  offsetY={0}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "right",
      "bottom",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 7`] = `
<Popover
  contentPosition={
    Array [
      "center",
      "top",
    ]
  }
  mode="hover"
  offsetX={0}
  offsetY={10}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "center",
      "bottom",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 8`] = `
<Popover
  contentPosition={
    Array [
      "left",
      "top",
    ]
  }
  mode="hover"
  offsetX={0}
  offsetY={10}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "left",
      "bottom",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 9`] = `
<Popover
  contentPosition={
    Array [
      "right",
      "top",
    ]
  }
  mode="hover"
  offsetX={0}
  offsetY={10}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "right",
      "bottom",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 10`] = `
<Popover
  contentPosition={
    Array [
      "right",
      "center",
    ]
  }
  mode="hover"
  offsetX={-10}
  offsetY={0}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "left",
      "center",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 11`] = `
<Popover
  contentPosition={
    Array [
      "right",
      "top",
    ]
  }
  mode="hover"
  offsetX={-10}
  offsetY={0}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "left",
      "top",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;

exports[`Tooltip renders correctly 12`] = `
<Popover
  contentPosition={
    Array [
      "right",
      "bottom",
    ]
  }
  mode="hover"
  offsetX={-10}
  offsetY={0}
  onChange={[Function]}
  selector="body"
  triggerPosition={
    Array [
      "left",
      "bottom",
    ]
  }
>
  <PopoverTrigger
    disabled={false}
  >
    <span>
      trigger
    </span>
  </PopoverTrigger>
  <PopoverContent
    contentPosition={
      Array [
        "left",
        "top",
      ]
    }
    selector="body"
    triggerPosition={
      Array [
        "left",
        "bottom",
      ]
    }
  >
    <Component />
  </PopoverContent>
</Popover>
`;
